nixos/modules/services/monitoring/prometheus/exporters/minio.nix

```diff
@@ -0,0 +1,71 @@
+{ config, lib, pkgs }:
+
+with lib;
+
+let
+  cfg = config.services.prometheus.exporters.minio;
+in
+{
+  port = 9290;
+  extraOpts = {
+    minioAddress = mkOption {
+      type = types.str;
+      example = "https://10.0.0.1:9000";
+      description = ''
+        The URL of the minio server.
+        Use HTTPS if Minio accepts secure connections only.
+        By default this connects to the local minio server if enabled.
+      '';
+    } // optionalAttrs config.services.minio.enable {
+      default = "http://localhost:9000";
+    };
+
+    minioAccessKey = mkOption {
+      type = types.str;
+      example = "BKIKJAA5BMMU2RHO6IBB";
+      description = ''
+        The value of the Minio access key.
+        It is required in order to connect to the server.
+        By default this uses the one from the local minio server if enabled
+        and <literal>config.services.minio.accessKey</literal>.
+      '';
+    } // optionalAttrs config.services.minio.enable {
+      default = config.services.minio.accessKey;
```

Comment:
This should probably be set later in the config = ... bit instead, to avoid regenerating documentation.